### PR TITLE
[Config] Add RectorConfigBuilder

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Wow, this looks amazing :pray:  I'll review and merge today

**Comment by TomasVotruba** [comment]
LGTM, let's test this in the wild :rocket:

File: rector.php [hunk 1]
```diff
@@ -24,26 +25,20 @@
         SetList::EARLY_RETURN,
         SetList::CODING_STYLE,
         SetList::STRICT_BOOLEANS,
-    ]);
-
-    $rectorConfig->rules([DeclareStrictTypesRector::class, MoveAbstractRectorToChildrenRector::class]);
-
-    $rectorConfig->paths([
+    ])->withRules([DeclareStrictTypesRector::class, MoveAbstractRectorToChildrenRector::class])
```
Comment (Wirone) [inline rector.php:28]: My inner linter 😅 

```suggestion
])
->withRules([DeclareStrictTypesRector::class, MoveAbstractRectorToChildrenRector::class])
```

But anyway, I wanted to say it's probably good idea to have such builder, but why it was merged without tests 😅?

**Pull request merged**